draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2084,31 +2113,42 @@ unambiguously.
 When hashing to a twisted Edwards curve that does not have a standardized
```

Comment:
> When defining new twisted Edwards curves, a Montgomery equivalent and rational
> map SHOULD be specified, and the sign of the rational map SHOULD be stated
> unambiguously.

According to the last modifications, this paragraph should be updated too with something like this:

> When defining new twisted Edwards curves, a **Weierstrass** equivalent and rational
> map SHOULD be specified, and the sign of the rational map SHOULD be stated
> unambiguously.

Reply:
This is outside the scope of this PR, so I think we should fix it elsewhere---if we do. But:

Note that this section is (and has been since we included it) a discussion of Edwards-to-Montgomery, not Edwards-to-Weierstrass. So what is written here *is* consistent with this section.

On the other hand, you make a good point: maybe this is needlessly indirect! Why not just do Edwards-to-Weierstrass and then use Elligator2 directly? From my perspective, the answer is that most curves will have an Edwards-to-Montgomery map defined, and thus most readers will already be familiar with that map---so talking about it in those terms will be more familiar to the reader.

Also, remember: the reason we do Edwards-to-Montgomery in the first place is so that Edwards and Montgomery hashes are compatible, *and can share an implementation*. So the whole point is, once someone has a hash to a Montgomery curve, they just implement a map to the Edwards and they're done.

Finally, in practice, most Montgomery curves that people use have B = 1, which means that they're already equivalent to the long Weierstrass form---so this distinction is mostly academic. But from my perspective it's cleaner to keep the internal details of the Montgomery map in the Montgomery section, and treat it as a black box here.

If you'd like to keep discussing this, maybe let's move it to its own issue?

Reply:
Move discussion to 195